backend/inspirehep/submissions/views.py

```diff
@@ -158,6 +166,75 @@ def create_ticket(self, record, rt_template):
         )
 
 
+class SeminarSubmissionsResource(BaseSubmissionsResource):
+    def post(self):
+        """Adds new conference record"""
+
+        data = self.load_data_from_request()
+
+        data['acquisition_source'] = self.get_acquisition_source()
+        record = SeminarsRecord.create(data)
+        db.session.commit()
+        return (jsonify({"pid_value": record["control_number"]}), 201)
+
+    def get(self, pid_value):
+        try:
+            record = SeminarsRecord.get_record_by_pid_value(pid_value)
+        except PIDDoesNotExistError:
+            abort(404)
+
+        serialized_record = seminar_serializer_v1.dump(record)
+
+        return jsonify({"data": serialized_record})
+
+    def put(self, pid_value):
+        try:
+            record = SeminarsRecord.get_record_by_pid_value(pid_value)
+            if not self.user_can_edit(record):
+                return (
+                    jsonify({"message": "You are not allowed to edit this seminar"}),
+                    403,
+                )
+        except PIDDoesNotExistError:
+            abort(404)
+        data = self.load_data_from_request()
+        builder = self.get_builder_with_updated_record(data, record)
+        record.update(builder.record)
+        db.session.commit()
+
+        return jsonify({"pid_value": record["control_number"]})
+
+    def get_builder_with_updated_record(self, data, record):
+        deletable_fields = [
+            "address",
+            "series",
+            "contact_details",
+            "urls",
+            "join_urls",
+            "abstract",
+            "keywords",
+            "public_notes",
+        ]
+
+        record_data = dict(record)
+        for field in deletable_fields:
+            if field not in data and field in record_data:
+                del record_data[field]
+        record_data.update(data)
+        builder = SeminarBuilder(record=record_data)
```

Comment:
do we neeed the builder here? Can we use the serializer?

Reply:
leftover! btw we really need to check other views and simplify them similar to Seminars